lib/puppet/util/log.rb

```diff
@@ -184,6 +184,7 @@ def Log.newmessage(msg)
     return if @levels.index(msg.level) < @loglevel
 
     msg.message = coerce_string(msg.message)
+    msg.source = coerce_string(msg.source)
```

Comment:
I opted to only do `source`.  Should I also do `file`, `environment`, `node` or `tags` @MosesMendoza?

Reply:
Probably not, if I'm correct in thinking that none of those could be anything other than UTF-8?

Reply:
I think `file` could plausibly be `Encoding.find(:filesystem)`... which might not be UTF-8... but that would require some real effort on a user to install Puppet (or set the module dir for instance) to a path containing Unicode characters.

I'm inclined to punt on that for now.